src/util/kaldi-holder-inl.h

```diff
@@ -101,6 +102,12 @@ template<class KaldiType> class KaldiObjectHolder {
     std::swap(t_, other->t_);
   }
 
+  bool ExtractRange(KaldiObjectHolder<T> *other, std::string range) {
```

Comment:
It would be more efficient to make all of these 'range' arguments const std::string&.